### ISSUE TITLE
add a "module" setting to package.json

### ISSUE THREAD

**Issue by giniedp** [quilljs/quill#2473]
Its my second attempt to use quill in a webpack/typescript/angular setup and i am having problems getting it to work with proper typings support. I am reading issues like 
https://github.com/quilljs/quill/issues/777 and https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18946 both are closed but without any proper solution.

I do compile to "commonjs" and i can not switch to "es2015" as it is suggested in https://github.com/quilljs/quill/issues/1814 
I think quill should not dictate my compilation target.

To help all the people that are using a bundler like webpack or rollup it is sufficient just to add the following line into the quills `package.json`

```json
   "module": "quill.js"
```

During compilation the bundler will then pick up that `quill.js` file instead of the precompiled `dist/quill.js` and then it all works as expected.

I would love to see quill being implemented in typescript so it can distribute its own d.ts files without us having to rely on @types/quill. In this case we then can have our custom implementation of quill.js while enjoying full typings support

**Comment by giniedp** [comment]
For now i use an alias when compiling with webpack. My webpack config looks like this

```
{
  resolve: {
    alias: { quill: path.join(process.cwd(), "node_modules", "quill", "quill.js") }
  }
  // ...
}
```

I get the typings for the "quill" library from "@types/quill" and can use quill import like

```ts
import Quill from "quill"
```
or

```ts
import { default as Quill, DeltaOperation } from "quill"
```

But now i want to make use of the Delta. I have to import it like

```ts
import * as Delta from "quill-delta"
```

here i don't have typings.

I see that the "parchment" is already implemented in typescript. Are you going to rewrite all quill libraries in typescript? This would be great for the whole project. I offer my help if you need it.

**Comment by luin** [comment]
I think this should have been solved in 2.0.0-beta.0. It also provides the official TypeScript typings.
